internal/exec/util/blkid.c

```diff
@@ -15,9 +15,11 @@
 #include <blkid/blkid.h>
 #include "blkid.h"
 
-result_t filesystem_type(const char *device, char type[], size_t type_len)
+result_t filesystem_type(const char *device, char type[], size_t type_len, char uuid[], size_t uuid_len, char label[], size_t label_len)
```

Comment:
I don't like overloading this function. It should either be renamed and use a struct or split into multiple functions.